File: src/main/java/com/koddy/server/auth/domain/model/code/DefaultAuthCodeGenerator.java
package com.koddy.server.auth.domain.model.code;

import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
public class DefaultAuthCodeGenerator implements AuthCodeGenerator {
    @Override
    public String get() {
        return UUID.randomUUID().toString().replaceAll("-", "").substring(0, 8);
    }
}

File: src/main/java/com/koddy/server/auth/domain/model/code/DefaultAuthCodeGenerator.kt
package com.koddy.server.auth.domain.model.code

import org.springframework.stereotype.Component
import java.util.UUID

@Component
class DefaultAuthCodeGenerator : AuthCodeGenerator {
    override fun get(): String {
        return UUID.randomUUID()
            .toString()
            .replace("-".toRegex(), "")
            .substring(0, 6)
    }
}
